packages/database/src/TupaiaDatabase.js

```diff
@@ -553,6 +564,16 @@ function addWhereClause(connection, baseQuery, where) {
       const { sql = value, parameters } = value;
       return querySoFar.whereRaw(sql, parameters);
     }
+    if (key === WHERE_SUBQUERY_CLAUSES.EXISTS) {
+      return querySoFar.whereExists(function () {
+        this.query(value);
+      });
+    }
+    if (key === WHERE_SUBQUERY_CLAUSES.NOT_EXISTS) {
+      return querySoFar.whereNotExists(function () {
+        this.query(value);
+      });
+    }
```

Comment:
I was pretty shocked to find this code worked perfectly right off the bat 🎉 I find this area of code to be pretty complex, and the lack of types really makes it hard to reason about. But on the flip side the api itself is super elegant, so doing something crazy and recursive like this just worked!